### ISSUE TITLE
State of HTML 2025 Suggestions

### ISSUE THREAD

**Issue by SachaG** [Devographics/surveys#253]
### Browser Interoperability Opinion Question

Current question:

> The lack of browser interoperability is a major obstacle to web development

Issue: people might interpret this as "[when there is a] lack of browser interoperability [it] is a major obstacle to web development" and check "agree" whether or not it's something that they experience right now. 

Reword it to: 

> The lack of browser interoperability is a major obstacle to web development today

**Comment by SachaG** [comment]
Clarify the term "validation" to avoid confusion with form validation: 

> I’ve moved to speaking of “conformance checking,” and only to speak of
anything valid in the context of the output being (in)valid HTML or
CSS. While it has the problem that not everyone makes the connection
to HTML validation (an expression that may be an alternative, too, and
which I’d prefer over “DOM validation”), it seems to prevent being
mistaken for input validation.

**Comment by SachaG** [comment]
Ask about devtools? (which features people use/wish were better?)

**Comment by SachaG** [comment]
https://github.com/whatwg/dom/issues/1255

**Comment by SachaG** [comment]
- alternative text for generated content: https://adrianroselli.com/2020/10/alternative-text-for-css-generated-content.html

**Comment by davatron5000** [comment]
Web Components section is still useful. Fine to slim things down. One question I'm curious about and I wonder if this bears out in the data... is the "7% dislike" statistic a majority of people who tried them and never looked back? Or is it people who are frequent users and care about the technology but are being harsh critics?

**Comment by SachaG** [comment]
> Compared to the past, do you think new web features are becoming stable across major browsers faster today, slower today, or at about the same pace?

I think this might be a bit too vague since "the past" is very subjective…

> How satisfied are you with the speed at which new web features and APIs become reliably usable across major web browsers?

I also don't think we necessarily need both questions, they seem to more or less be asking the same thing to me? 

To review, this is the question we have currently:

> The lack of browser interoperability is currently a major obstacle to web development (disagree/agree/etc.)

If you want to measure people's thought on the speed of improvement, we could adapt your second question to fit the same format and replace it with:

> New web features and APIs take too much time to become reliably usable across major web browsers (disagree/agree/etc.)

**Comment by LeaVerou** [comment]
Just a quick note that after taking the input here into consideration (and the community input that followed https://lea.verou.me/blog/2025/design-state-of-html/ ) @SachaG and I have been fleshing out the outline [here](https://coda.io/d/State-of-HTML-2025_djmOm2j3kjR/Research_sulETNyd#_luClg0oJ). It's already shared with @atopal and @foolip, if any other stakeholders want to have a more involved role in fleshing out the outline feel free to request access!

**Comment by LeaVerou** [comment]
> Thank you [@LeaVerou](https://github.com/LeaVerou)! I see that `ctx.drawElement()` (HTML in canvas) is in there and that was my main request. [@progers](https://github.com/progers) [@schenney](https://github.com/schenney) FYI

I’m wondering about adding a common pain point around this too, e.g. "Rending rich text on canvas is painful".

Btw the survey preview is [here](https://survey.devographics.com/en-US/survey/state-of-html/2025/), though we're still fleshing out several things (common pain points + re-adding a Content section are the main ones, the rest are small tweaks).

**Comment by foolip** [comment]
@SachaG it was on Chrome 138.0.7204.101 on macOS 15.5 (24F74). It seems like it's broken between two breakpoints, from width ~990 until width ~1200. Narrower or wider than that range it's fine.
